llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -16485,6 +16485,34 @@ static SDValue expandMulToAddOrSubOfShl(SDNode *N, SelectionDAG &DAG,
   return DAG.getNode(Op, DL, VT, Shift1, Shift2);
 }
 
+static SDValue getShlAddShlAdd(SDNode *N, SelectionDAG &DAG, int ShX, int ShY) {
```

Comment:
Can we use `unsigned` for `ShX/ShY` since they can't be negative? If they could be negative then `getConstant` below needs to be `getSignedConstant`. So I'd like the code to be self consistent.